signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerFactory.java

```diff
@@ -66,8 +67,14 @@ public Signer createSigner(final AzureConfig config) {
       throw new SignerInitializationException(INVALID_KEY_PARAMETERS_ERROR, e);
     }
     final JsonWebKey jsonWebKey = cryptoClient.getKey().getKey();
+    final String curveName = jsonWebKey.getCurveName().toString();
+    if (!curveName.equals("SECP256K1") && !curveName.equals("P-256K")) {
+      LOG.error("Unsupported curve name: {}. Expecting P-256K", curveName);
+      throw new SignerInitializationException(INVALID_CURVE_NAME);
+    }
     final Bytes rawPublicKey =
         Bytes.concatenate(Bytes.wrap(jsonWebKey.getX()), Bytes.wrap(jsonWebKey.getY()));
-    return new AzureKeyVaultSigner(config, rawPublicKey, needsToHash);
+    return new AzureKeyVaultSigner(
+        config, rawPublicKey, needsToHash, curveName.equals("SECP256K1"));
```

Comment:
extracting curveName.equals("SECP256K1") as a variable say useDeprecated would make this clearer